apps/barista-design-system/src/app/app.ts

```diff
@@ -0,0 +1,165 @@
+]);
+
+export const DEFAULT_PAGE_THEME = 'turquoise';
+
+export interface BaBreadCrumb {
+  label: string;
+  url: string;
+}
+
+@Component({
+  selector: 'ba-app',
+  templateUrl: './app.html',
+  styleUrls: ['./app.scss'],
+})
+export class BaApp implements OnInit, OnDestroy {
+  _url$ = this._router.events.pipe(
+    filter((event: Event) => event instanceof NavigationEnd),
+    pluck('url'),
+    distinctUntilChanged(),
+    share(),
+  );
+
+  /** The stream that holds the current breadcrumbs */
+  _breadcrumbs$ = this._url$.pipe(
+    map((url: string) => this._createBreadcrumbs(url)),
+  );
+
+  private _destroy$ = new Subject<void>();
+
+  constructor(
+    private _pageService: BaPageService,
+    private _router: Router,
+    private _meta: Meta,
+    private _title: Title,
+    @Inject(DOCUMENT) private _document: any,
+  ) {}
+
+  ngOnInit(): void {
+    //Called after the constructor, initializing input properties, and the first call to ngOnChanges.
+    //Add 'implements OnInit' to the class.
+    this._url$
+      .pipe(
+        map((url: string) => {
+          const key = getPageKeyFromUrl(this._document, url);
+          return this._pageService._cache.get(key);
+        }),
+        takeUntil(this._destroy$),
+      )
+      .subscribe(page => {
+        this._setMeta(page);
+      });
+  }
+
+  ngOnDestroy(): void {
+    this._destroy$.next();
+    this._destroy$.complete();
+  }
+
+  /** @internal Gets the page theme based on the current location. */
+  _getPageTheme(): string {
+    const path = this._router.url.substr(1);
+    let pageTheme = DEFAULT_PAGE_THEME;
+    if (path.length) {
+      const firstPart = path.split('/')[0];
+      pageTheme = PAGE_THEME_MAP.get(firstPart) || pageTheme;
+    }
+    return pageTheme;
+  }
+
+  /** @internal Generate Breadcrumbs based on the url */
+  _createBreadcrumbs(url: string): BaBreadCrumb[] {
+    const urlParts = getUrlPathName(this._document, url).split('/');
+
+    return urlParts.reduce((prev: BaBreadCrumb[], cur: string) => {
+      const currentUrl = `${prev.map(part => part.url).join('/')}/${cur}`;
+      const page = this._pageService._cache.get(
+        getPageKeyFromUrl(this._document, currentUrl),
+      );
+      const label = page ? page.title : cur.replace(/\-/gm, '');
+
+      prev.push({
+        label,
+        url: currentUrl,
+      });
+      return prev;
+    }, [] as BaBreadCrumb[]);
+  }
+
+  private _setMeta(page: BaContentTypes): void {
+    const pageTitle =
+      page.layout === BaPageLayoutType.Index
+        ? page.title
+        : `Barista - ${page.title}`;
+
+    const keywords = [
+      'Dynatrace',
+      'Barista',
+      'Design System',
+      'Open Source',
+      ...(page.hasOwnProperty('tags') ? page['tags'] : []),
+    ];
+
+    const metaTags: MetaDefinition[] = [
+      { property: 'og:url', content: this._router.url },
+      { property: 'og:title', content: pageTitle },
+      { property: 'keywords', content: keywords.join(', ') },
```

Comment:
could we add `description` here as well?

Reply:
if you look some lines below there it is:

```typescript
    if (page.description) {
      metaTags.push({ name: 'description', content: page.description });
    }
```

Reply:
whoops, sorry missed that one